noir/noir-repo/acvm-repo/brillig_vm/src/black_box.rs

```diff
@@ -38,6 +38,25 @@ pub(crate) fn evaluate_black_box<Solver: BlackBoxFunctionSolver>(
     bigint_solver: &mut BigIntSolver,
 ) -> Result<(), BlackBoxResolutionError> {
     match op {
+        BlackBoxOp::AES128Encrypt { inputs, iv, key, outputs } => {
+            let bb_func = black_box_function_from_op(op);
+
+            let inputs = to_u8_vec(read_heap_vector(memory, inputs));
+
+            let iv: [u8; 16] = to_u8_vec(read_heap_array(memory, iv)).try_into().map_err(|_| {
+                BlackBoxResolutionError::Failed(bb_func, "Invalid iv length".to_string())
+            })?;
+            let key: [u8; 16] =
+                to_u8_vec(read_heap_array(memory, key)).try_into().map_err(|_| {
+                    BlackBoxResolutionError::Failed(bb_func, "Invalid ley length".to_string())
+                })?;
```

Comment:
`to_u8_array` would avoid the need for these errors.

Reply:
uhm, I think you're referring to the method on `pwg/blackbox/utils`, but that operates over witnesses and I don't think is applicable here. I'm going with that's being done for `EcdsaSecp256r1` to assert the length of the incoming inputs.

Reply:
Ah, right you are.